server/rest/src/main/java/org/infinispan/rest/resources/ResourceUtil.java

```diff
@@ -0,0 +1,59 @@
+package org.infinispan.rest.resources;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Util class for REST resources.
+ *
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+class ResourceUtil {
+   private final static Log logger = LogFactory.getLog(ResourceUtil.class, Log.class);
+
+   static CompletableFuture<RestResponse> notFoundRspFuture() {
```

Comment:
```suggestion
   static CompletableFuture<RestResponse> notFoundResponseFuture() {
```

Reply:
And actually shouldn't this method store a cached response object to not allocate it every time? And this should probably return a CompletionStage<RestResponse> ?

Reply:
Actually looking further all of the methods here have the reduced `Rsp` wording. It personally feels a bit weird to me, but if you think it makes the methods too long I  guess I am fine with it. Also all of the methods should return `CompletionStage<..>`

Reply:
RestResponses should not be reused as they contain a netty byte buffer inside

Reply:
Ah, that is a shame because this response to me should be completely stateles...

Reply:
That is really disappointing, because to me this response should be completely stateless.. :(

Reply:
We tried it a while ago and the result was not good :)

Reply:
https://github.com/infinispan/infinispan/pull/7121

Reply:
> Ah, you could always just increment the refCount once and then it would never be collected though

I can't remember if I tried that or not, I remember it was tricky to track it down. Maybe you can give a try sometime?

Reply:
It probably isn't worth it.